addons/website_crm/models/crm_lead.py

```diff
@@ -7,7 +7,39 @@
 class Lead(models.Model):
     _inherit = 'crm.lead'
 
-    visitor_id = fields.Many2one('website.visitor', "Web Visitor")
+    visitor_ids = fields.Many2many('website.visitor', string="Web Visitors")
+    visitor_page_count = fields.Integer('# Page Views', compute="_compute_visitor_page_count")
+
+    def _compute_visitor_page_count(self):
+        self.flush(['visitor_ids'])
+        sql = """ SELECT l.id as lead_id, count(*) as page_view_count
+                    FROM crm_lead l
+                    JOIN crm_lead_website_visitor_rel lv ON l.id = lv.crm_lead_id
+                    JOIN website_visitor v ON v.id = lv.website_visitor_id
+                    JOIN website_visitor_page p ON p.visitor_id = v.id
+                    WHERE l.id in %s
+                    GROUP BY l.id"""
+        self.env.cr.execute(sql, (tuple(self.ids),))
+        page_data = self.env.cr.dictfetchall()
+        mapped_data = {data['lead_id']: data['page_view_count'] for data in page_data}
+        for lead in self:
+            lead.visitor_page_count = mapped_data.get(lead.id, 0)
+
+    def action_redirect_to_page_views(self):
+        self.ensure_one()
```

Comment:
Ensure one is not necessary, you would have the same action for a true multi record set if you want. isn't it ?